src/main/java/org/apache/maven/buildcache/xml/CacheConfigImpl.java

```diff
@@ -597,4 +597,15 @@ private String getProperty(String key, String defaultValue) {
         }
         return value;
     }
+
+    private boolean getProperty(String key, boolean defaultValue) {
+        String value = session.getUserProperties().getProperty(key);
+        if (value == null) {
+            value = session.getSystemProperties().getProperty(key);
+            if (value == null) {
+                return defaultValue;
+            }
+        }
+        return Boolean.parseBoolean(value);
```

Comment:
I think with system properties this is redundant, no?

Reply:
This could be accomplished any number of ways, but I decided to go with the micro-optimized form that does not require a string parse to handle the default case.